protocol_driver_grpc_async_callback.cc

```diff
@@ -316,4 +323,177 @@ void ProtocolDriverGrpcAsyncCallback::ShutdownServer() {
   }
 }
 
+namespace {
+class CallData {
+ public:
+  CallData(Traffic::AsyncService* service, grpc::ServerCompletionQueue* cq,
+           std::function<std::function<void()>(ServerRpcState* state)>* handler,
+           DistbenchThreadpool* thread_pool)
+      : service_(service),
+        cq_(cq),
+        handler_(handler),
+        responder_(&ctx_),
+        status_(CREATE) {
+    CHECK(thread_pool);
+    thread_pool_ = thread_pool;
+    ProcessRpcFsm();
+  }
+
+  // Calldata can be deleted from two places, whichever is last.
+  // 1. In FINISH state of ProcessRpcFsm()
+  // 2. From inside of rpc_state.SetFreeStateFunction()
+  void Unref() {
+    if (--refcnt_ == 0) {
```

Comment:
optional: can use atomic_fetch_sub to avoid adding a full memory barrier here implicitly.